opus/application/static_media/js/hash.js

```diff
@@ -45,6 +46,67 @@ var o_hash = {
         return hash.join("&");
     },
 
+    encodeSlugValue: function(slugValueArray) {
+        /**
+         * Take in a slug value array (like opus.selections, each element
+         * will be a value for the slug) and encode all values in the
+         * array. Return an array that contains encoded values for the
+         * slug. This function will be called in updateHash to make sure
+         * slug values in the hash are all encoded before updating the URL.
+         */
+        let slugValue = [];
+        $.each(slugValueArray, function(index, val) {
+            let value = encodeURIComponent(val);
+            value = value.replace(/\%20/g, "+");
+            slugValue.push(value);
+        });
+
+        return slugValue;
+    },
+
+    encodeHashArray: function(hashArray) {
+        /**
+         * Take in a hash array (each element will be "slug=value") and
+         * encode the "value" of each element. Return a hash array that
+         * is consist of "slug=encodedValue". This function will be called
+         * in opus.normalizedURLAPICall to make sure slug values from
+         * new URL are decoded.
+         */
+        let hash = [];
+        $.each(hashArray, function(index, pair) {
+            let slug = pair.split('=')[0];
+            let value = pair.split('=')[1];
+            value = encodeURIComponent(value);
```

Comment:
So we have this legacy code everywhere that assumes values (and qtypes) are *lists*, not just single values. This is because the old version of OPUS allowed multiple search values for a given field, e.g. "observationduration1=1,2,3" and "observationduration2=4,5,6". We might actually want to allow this again at some point, so we might as well support it everywhere. It seems like the better thing to do here, rather than just do a single encodeURIComponent, is to split the value at ",", then call encodeSlugValue, and then join the results back with "," again. This also means the encode is only done in one place, so we don't have duplicate code that can diverge over time.